### ISSUE TITLE
Unable to connect to notebook that is already running

### ISSUE THREAD

**Issue by olaviinha** [googlecolab/colabtools#2492]
**Describe the current behavior**
I left a notebook running on computer no. 1. I am unable to access that running instance and thus unable to use the notebook at all on computer no. 2.

![image](https://user-images.githubusercontent.com/50331907/145539536-6753ce35-7fad-4a5f-bd30-2a309cd18753.png)

It just keeps saying "Connecting", but never connects. 

Browser console reveals it keeps connecting and disconnecting approximately every 30 seconds on computer no. 2. This seems to be an endless loop:
```
10:23:04.656 Connected to https://colab.research.google.com/tun/m/gpu-p100-s-***
10:24:29.956 Runtime disconnected
10:24:33.687 Connected to https://colab.research.google.com/tun/m/gpu-p100-s-***
10:25:57.934 Runtime disconnected
```

I have experienced this multiple times in recent times. 

**Describe the expected behavior**
I should be able to use the notebook (and connect to its runtime) on computer no. 2, even if it's already running, to e.g. stop it.

It used to work just fine in the past, not so long ago: when you opened the running notebook on another computer, it would connect to the hosted runtime where it's already running. So you could switch computer on the fly without interrupting the notebook – super convenient btw!

**What web browser you are using**
Chrome (on both computers)

**Additional context**
- I have Pro subscription (no background execution). 
- I know that connecting to an already running notebook can take a few minutes (on Pro+ plan, which I would still prefer if only it gave GPU access, mind you), so I can confirm that this is not the case here. It still keeps "connecting" after waiting for hours.

**Comment by blois** [comment]
You should be able to reconnect to the same runtime from different computers. The most common cause of the 'Connecting' hanging is the runtime getting into a bad state.

Are you able to repro this?

**Comment by olaviinha** [comment]
I can confirm that the notebook is still running just fine even when I can no longer access it. It continues to save output to my Drive as usual.

And yes, in fact I can reproduce it even without a second computer (noticed after reporting this bug). I had a notebook with a cell running just now. I hit refresh on chrome, and now the notebook is in this "Connecting" state. Same thing happening in browser console: keeps connecting and disconnecting.

**Comment by olaviinha** [comment]
I just tested this with another notebook, and this problem did not occur, so problem appears to occur only in one particular notebook. It has very heavy and truncated cell outputs though, that's the only difference I can think of. If you have any tips on what is a _bad state_, that would help me troubleshoot.

**Comment by blois** [comment]
The stalled at connecting typically means that the connection was established to the kernel but the kernel was unresponsive. Particularly large outputs could potentially cause this though it should be quite hard. If you can repro it we'd be interested in what you find. More often I've seen critical dependencies be upgraded to incompatible versions but that shouldn't impact reconnect.

Typically issues that cause the runtime to become unresponsive will cause Colab to consider the VM unhealthy and recycle it but this is quite rare.

**Comment by olaviinha** [comment]
I did some more testing and I'm now thinking it was indeed related to super large cell outputs on that particular notebook. After adding a `output.clear()` to run every now and then, problem disappeared. Tested also with other notebooks with light outputs, and problem did not occur.

I quickly tried to calculate just now the size of output in the problematic notebook, and I think it was being several gigabytes. I can see how that _does not compute_.

I feel like closing the issue because notebook itself was rather sour, but will leave it up to you, because basically this bug still exists and is reproducible with super large cell outputs. I think the notebook will not display the entire output even after session termination, so I guess that could already be done upon reconnecting to runtime: if output is too large, just force-clear the output and display a message, similar to the truncation message.
